File: array-misc/given-a-set-a-of-n-numbers-and-another-number-x-determines-whether-or-not-there-exist-two-elements-in-s-whose-sum-is-exactly-x.cpp
#include <cstdio>
#define MAX 100000

using namespace std;
 
void printPairs(int arr[], int arr_size, int sum)
{
  int i, temp;
  bool binMap[MAX] = {0}; /*initialize hash map as 0*/
 
  for (i = 0; i < arr_size; i++)
  {
      temp = sum - arr[i];
      if (temp >= 0 && binMap[temp] == 1)
         printf("Pair with given sum %d is (%d, %d) \n", 
                 sum, arr[i], temp);
      binMap[arr[i]] = 1;
  }
}
 
/* Driver program to test above function */
int main()
{
    int A[] = {2, 4, 1, 5};
    int n = 6;
    int arr_size = sizeof(A)/sizeof(A[0]);
 
    printPairs(A, arr_size, n);

    return 0;
}

File: array-misc/given-a-set-a-of-n-numbers-and-another-number-x-determines-whether-or-not-there-exist-two-elements-in-s-whose-sum-is-exactly-x.py
# Python program to find if there are two elements wtih given sum
CONST_MAX = 100000
 
# function to check for the given sum in the array
def printPairs(arr, arr_size, sum):
     
    # initialize hash map as 0
    binmap = [0]*CONST_MAX
     
    for i in range(0,arr_size):
        temp = sum-arr[i]
        if (temp>=0 and binmap[temp]==1):
            print ("Pair with the given sum is", arr[i], "and", temp)
        binmap[arr[i]]=1
 
# driver program to check the above function
A = [2, 4, 1, 5]
n = 6
printPairs(A, len(A), n)